components/user-info/frontend/src/view/change-password/change-password-dialog.html

```diff
@@ -0,0 +1,49 @@
+<div class="edit-user-info-form">
+  <p translate>user-info.password-change.introduction</p>
+  <form class="form-stacked center-block" name="form.changePasswordForm" role="form">
+
+    <div class="form-group has-feedback">
+      <label class="control-label" for="user-info-form-oldPassword" translate>
+        user-info.password-change.old
+      </label>
+      <input autofocus type="password" password-reveal required class="form-control"
+        id="user-info-form-oldPassword" name="oldPassword"
+        placeholder="{{ 'user-info.password-change.placeholders.old' | translate }}"
+        ng-model="asyncTaskDialogCtrl.context.data.oldPassword"/>
+    </div>
+
+    <div class="form-group has-feedback">
+      <label class="control-label" for="user-info-form-newPassword" translate>
+        user-info.password-change.new
+      </label>
+      <input type="password" password-reveal required class="form-control"
+        id="user-info-form-newPassword" name="newPassword"
+        placeholder="{{ 'user-info.password-change.placeholders.new' | translate }}"
+        ng-model="asyncTaskDialogCtrl.context.data.newPassword"/>
+    </div>
+
+    <div class="form-group has-feedback"
+      ng-class="{'has-error': asyncTaskDialogCtrl.context.data.newPassword !== asyncTaskDialogCtrl.context.data.newPasswordConfirm}">
+      <label class="control-label" for="user-info-form-newPasswordConfirm" translate>
+        user-info.password-change.confirm
+      </label>
+      <span class="control-message" ng-if="asyncTaskDialogCtrl.context.data.newPassword !== asyncTaskDialogCtrl.context.data.newPasswordConfirm">
```

Comment:
If you wrap these inputs in a form with name 'form.<something>' (syntax important) you can then access the $dirty property to only show the warning once the user has input text. This all ties in to the validity check the dialog does. I think if if you then use an ng-pattern validator here to check for the password equality you could remove the custom isFormInvalid function in change-password.service.js